### PR TITLE
Fix NNVisualizer

### PR DESCRIPTION
**Pull request by ofirgo**
KerasNNVisualizer was imported from common (bug). Now,
post_training_quantization uses NNVisualizer from common
and passes framework objects to be able to use it in PyTorch as well.
Also, changed similarity visualizer to work with given distance method (instead of default cs).
TestLogger was modified to use analyze_similarity flag to test it.

### CODE REVIEW

File: tests/keras_tests/function_tests/test_tensorboard_writer.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ class TestLogger(unittest.TestCase):
     def setUpClass(cls):
         mct.core.common.Logger.set_log_file('/tmp/')
         model = MobileNet()
-        mct.keras_post_training_quantization(model, random_datagen, n_iter=1)
+        mct.keras_post_training_quantization(model, random_datagen, n_iter=1, analyze_similarity=True)
```
Comment (reuvenperetz) [inline tests/keras_tests/function_tests/test_tensorboard_writer.py:39]: Please use this flag in https://github.com/sony/model_optimization/blob/main/tests/pytorch_tests/function_tests/test_tensorboard_writer.py as well.

**Review by reuvenperetz — CHANGES_REQUESTED** [review]

File: model_compression_toolkit/core/common/constants.py [hunk 2]
```diff
@@ -41,7 +41,7 @@
 
 
 # Number of Tensorboard cosine-similarity plots to add:
-NUM_SAMPLES_CS_TENSORBOARD = 20
+NUM_SAMPLES_DISTANCE_TENSORBOARD = 20
```
[the remark this answers is not in the document]
Comment (ofirgo) [reply model_compression_toolkit/core/common/constants.py:45]: The default (in the quantization facade) is still cosine similarity distance, so I'm not sure that we have something to update here, the example is still valid.

[the remark this answers is not in the document]
Comment (reuvenperetz) [reply model_compression_toolkit/core/common/constants.py:45]: But the image has different axis labels, and maybe we should mention that it is possible to display different metrics (but it's CS by default).
